src/devices/GrovePi/Sensors/LedBar.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Iot.Device.GrovePiDevice.Models;
+
+namespace Iot.Device.GrovePiDevice.Sensors
+{
+    /// <summary>
+    /// The led bar orientation
+    /// </summary>
+    public enum LedBarOrientation
+    {
+        RedToGreen = 0,
+        GreenToRed = 1
+    }
+
+    /// <summary>
+    /// LedBar class to support grove LedBar http://wiki.seeedstudio.com/Grove-LED_Bar/
+    /// </summary>
+    public class LedBar : ISensor<byte>
+    {
+        private GrovePi _grovePi;
+        private byte _level;
+        private LedBarOrientation _orientation;
+
+        /// <summary>
+        /// LedBar constructor
+        /// </summary>
+        /// <param name="grovePi">The GrovePi class</param>
+        /// <param name="port">The grove Port, need to be in the list of SupportedPorts</param>
+        public LedBar(GrovePi grovePi, GrovePort port) : this(grovePi, port, LedBarOrientation.GreenToRed)
+        { }
+
+        /// <summary>
+        /// LedBar constructor
+        /// </summary>
+        /// <param name="grovePi">The GrovePi class</param>
+        /// <param name="port">The grove Port, need to be in the list of SupportedPorts</param>
+        /// <param name="orientation">Orientation, Green to red is default</param>
+        public LedBar(GrovePi grovePi, GrovePort port, LedBarOrientation orientation)
+        {
+            if (!SupportedPorts.Contains(port))
+                throw new ArgumentException($"Error: grove Port not supported");
+            _grovePi = grovePi;
+            Port = port;
+            _orientation = orientation;
+            _grovePi.WriteCommand(GrovePiCommands.LedBarInitialization, port, (byte)_orientation, 0);
+            _level = 0;
+        }
+
+        /// <summary>
+        /// Get/Set the Led bar orientation
+        /// </summary>
+        public LedBarOrientation Orientation
+        {
+            get { return _orientation; }
+
+            set
+            {
+                if (_orientation != value)
+                {
+                    _orientation = value;
+                    _grovePi.WriteCommand(GrovePiCommands.LedBarOrientation, Port, (byte)_orientation, 0);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get/Set the level from 0 (no led on) to 10 (all leds on)
+        /// </summary>
+        public byte Value
+        {
+            get { return _level; }
+
+            set
+            {
+                _level = Math.Clamp(value, (byte)0, (byte)10);
+                _grovePi.WriteCommand(GrovePiCommands.LedBarLevel, Port, _level, 0);
+            }
+        }
+
+        /// <summary>
+        /// Set one led
+        /// </summary>
+        /// <param name="led">The led number from 0 to 10</param>
+        /// <param name="status">true for on, false for off</param>
+        public void SetOneLed(byte led, bool status)
+        {
+            led = Math.Clamp(led, (byte)0, (byte)10);
```

Comment:
I'd rather throw in here since it likely would be unintentional to light non-existing led

Reply:
@Ellerbach I think this comment got missed

Reply:
yes, missed that one, good catch! Fixed